pwndbg/commands/sigreturn.py

```diff
@@ -0,0 +1,120 @@
+from __future__ import annotations
+
+import argparse
+import collections
+
+import pwndbg.color.context as C
+import pwndbg.color.memory as M
+import pwndbg.color.message
+import pwndbg.commands
+import pwndbg.gdblib.arch
+import pwndbg.gdblib.memory
+import pwndbg.gdblib.regs
+from pwndbg.lib.regs import amd64 as amd64_regset
+
+parser = argparse.ArgumentParser(description="Display the SigreturnFrame at the specific address")
+
+parser.add_argument(
+    "address", nargs="?", default=None, type=int, help="The address to read the frame from"
+)
+
+parser.add_argument(
+    "-a",
+    "--all",
+    dest="display_all",
+    action="store_true",
+    default=False,
+    help="Show all values in the frame in addition to common registers",
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+@pwndbg.commands.OnlyWithArch(["x86-64"])
+def sigreturn(address: int = None, display_all=False):
+    address = pwndbg.gdblib.regs.sp if address is None else address
+
+    arch_name = pwndbg.gdblib.arch.name
+    if arch_name == "x86-64":
+        sigreturn_x86_64(address, display_all)
+    else:
+        print(
+            pwndbg.color.message.error(f"sigreturn does not support the {arch_name} architecture")
+        )
+
+
+SIGRETURN_FRAME_SIZE_x86_64 = 256
+
+# Original registers layout from pwntools, modified below : https://github.com/Gallopsled/pwntools/blob/e4d3c82501c03de44458ae498a830fe66594f66d/pwnlib/rop/srop.py#L256
```

Comment:
1. Why do we modify it?
2. Can't we just pull those values out of pwntools? Pwntools is a Pwndbg dependency :)

cc: @Arusekk

Reply:
I can remove the comment - the initial values were find of from there, but I needed to add +8 to add all the offsets and swap the keys and values (and add the initial key,value pair ("&pretcode", 0)). We could remove the comment if its misleading, in this case.

Reply:
Nah, lets leave the comment.

I was thinking more about just using this from here:
```
In [9]: pwnlib.rop.srop.SigreturnFrame().registers
Out[9]:
{0: 'uc_flags',
 8: '&uc',
 16: 'uc_stack.ss_sp',
 24: 'uc_stack.ss_flags',
 32: 'uc_stack.ss_size',
 40: 'r8',
 48: 'r9',
 56: 'r10',
 64: 'r11',
 72: 'r12',
 80: 'r13',
 88: 'r14',
 96: 'r15',
 104: 'rdi',
 112: 'rsi',
 120: 'rbp',
 128: 'rbx',
 136: 'rdx',
 144: 'rax',
 152: 'rcx',
 160: 'rsp',
 168: 'rip',
 176: 'eflags',
 184: 'csgsfs',
 192: 'err',
 200: 'trapno',
 208: 'oldmask',
 216: 'cr2',
 224: '&fpstate',
 232: '__reserved',
 240: 'sigmask'}
```
This also solves the problem with supporting different architectures - it just works out of the box as long as pwntools supports a given arch.